src/logistration/RegistrationPage.jsx

```diff
@@ -127,21 +236,97 @@ class RegistrationPage extends React.Component {
       nameValid,
       usernameValid,
       passwordValid,
-      countryValid,
     } = this.state;
+
+    const validityMap = REGISTRATION_VALIDITY_MAP;
+    const validStates = [];
+    Object.keys(validityMap).forEach((key) => {
+      const value = validityMap[key];
+      if (value) {
+        const state = camelCase(key);
+        const stateValid = `${state}Valid`;
+        validStates.push(stateValid);
+      }
+    });
+    let extraFieldsValid = true;
+    validStates.forEach((value) => {
+      extraFieldsValid = extraFieldsValid && this.state[value];
+    });
+
     this.setState({
-      formValid: emailValid && nameValid && usernameValid && passwordValid && countryValid,
+      formValid: emailValid && nameValid && usernameValid && passwordValid && extraFieldsValid,
     });
   }
 
-  renderCountryList() {
-    const locale = getLocale();
-    let items = [{ value: '', label: 'Country or Region of Residence (required)' }];
-    items = items.concat(getCountryList(locale).map(({ code, name }) => ({ value: code, label: name })));
-    return items;
+  addExtraFields() {
+    const fields = this.props.formData.fields.map((value) => {
+      const field = value;
+      if (field.required === true && REGISTRATION_EXTRA_FIELDS.includes(field.name)) {
+        const stateVar = camelCase(field.name);
```

Comment:
Instead of converting `field.name` in camel case at multiple places, we could simply convert the whole fields object into camel case using `import { camelCaseObject } from '@edx/frontend-platform';` at a single place.

Reply:
@waheedahmed camelCaseObject converts keys into camelCase. Here i need to convert the value. So i had to use the camelCase function directly.

<img width="769" alt="Screen Shot 2020-11-27 at 1 44 30 PM" src="https://user-images.githubusercontent.com/373677/100429224-d8897400-30b6-11eb-8ceb-067b44c95db5.png">

Reply:
Got it, thanks!